### DISCUSSION TITLE
determining boundary plane trace

### DISCUSSION THREAD

**Discussion by Saoussen-ouhiba** [mtex-toolbox/mtex#636] [Ask Anything] [answered]
Hello,

I have a 2D EBSD map and I would like to calculate the trace of each boundary plane and determine whether this trace could lie within the plane {111}.

Does someone have an idea about how to do it please?

Thank you a lot for your help,

Saoussen

**Comment by kilir** [comment, marked as answer]
Hi,
maybe something like that?
```
% some example data
mtexdata csl
ebsd= ebsd(inpolygon(ebsd,[0 0 200 200]));
grains = ebsd.calcGrains;
grains = grains(grains.grainSize>100);
grains = grains.smooth(15)
grains=grains('ir')
%% take all symmetrically equivalent 111
hsym = symmetrise(Miller(1,1,1,ebsd('ir').CS),'unique')

% choose a grains
i = 45

% traces of the hsym within the grain
tr = cross(grains(i).meanOrientation*hsym,zvector);

% minimum angle between gb direction and traces
gba = min(angle_outer(g.boundary('indexed').direction,tr),[],2);

plot(g.boundary('indexed'),gba/degree,'linewidth',2)
hold on
for j=1:length(hsym)
quiver(g, tr(j) ,'antipodal','color','r')
end
hold off
mtexColorbar
```

Cheers,
Rüdiger

**Reply by Saoussen-ouhiba** [reply]
Thank you so much Rüdiger. I tried to modify the script that you sent me to include all the grains in my EBSD map. The code that I wrote is below. 

``` 
% determine large grains
[grains_LLASS,ebsd_LLASS.grainId,ebsd_LLASS.mis2mean] = calcGrains(ebsd_LLASS('Aluminium'),'angle',10*degree);
large_grains_LLASS= grains_LLASS((grains_LLASS.GOS<3*degree) & (grains_LLASS.area>4000));
large_grains_smoothed = large_grains_LLASS.smooth(10)

% take all symmetrically equivalent 111
hsym = symmetrise(Miller(1,1,1,ebsd_LLASS('Al').CS),'unique')

% traces of the hsym within the grains
tr = cross(large_grains_smoothed.meanOrientation*hsym,zvector);

% minimum angle between gb direction and traces
for i =1:length(large_grains_smoothed)
dir = large_grains_smoothed(i).boundary('indexed').direction;
gba(1:length(dir),i) = min(angle_outer(dir,tr(i,:)),[],2);
end

figure
for k = 1:length(large_grains_smoothed)
gb = large_grains_smoothed(k).boundary('indexed');
plot(gb,gba(1:length(gb.direction),k)/degree,'linewidth',2)
hold on
for j=1:length(hsym)
quiver(large_grains_smoothed(k), tr(j) ,'antipodal','color','r')
end
end
hold off
mtexColorbar
```

Unfortunately, when I tested your code and mine for one grain for example large_grains_smoothed(3), I did not get the same result. Figure 1 shows the result that I get with my code and Figure 2 shows the result that I get with your script as given to me without modification:

![figure1](https://user-images.githubusercontent.com/73701277/102991335-4b9cd380-4519-11eb-914a-de06482b67ae.jpg)
![figure2](https://user-images.githubusercontent.com/73701277/102991348-4fc8f100-4519-11eb-9cf6-5c2e06fd7a10.jpg)

Could you tell me where is the problem in my code please ? Thank you a lot and sorry for bothering you again,

Saoussen

**Reply by kilir** [reply]
Since you have grains-by-"symmetrically equivalent {111}" it should be 
```
quiver(grains(k), tr(k,j) ,'antipodal','color','r')
```
Cheers,
Rüdiger

**Reply by Saoussen-ouhiba** [reply]
Thank you so much for your help. It is working now.
